yt_dlp/extractor/bandlab.py

```diff
@@ -0,0 +1,438 @@
+        'md5': 'a4ad05cb68c54faaed9b0a8453a8cf4a',
+        'info_dict': {
+            'id': '014de0a4-7d82-ea11-a94c-0003ffd19c0f',
+            'ext': 'm4a',
+            'comment_count': int,
+            'genres': ['Electronic'],
+            'uploader': 'Charlie Henson',
+            'timestamp': 1587328674,
+            'upload_date': '20200419',
+            'view_count': int,
+            'track': 'Positronic Meltdown',
+            'duration': 318.55,
+            'thumbnail': 'https://bandlabimages.azureedge.net/v1.0/songs/87165bc3-5439-496e-b1f7-a9f13b541ff2/',
+            'description': 'Checkout my tracks at AOMX http://aomxsounds.com/',
+            'uploader_id': 'microfreaks',
+            'title': 'Positronic Meltdown',
+            'like_count': int,
+            'media_type': 'revision',
+        },
+    }, {
+        # Track without revisions available
+        'url': 'https://www.bandlab.com/track/55767ac51789ea11a94c0003ffd1fc09_2f007b0a37b94ec7a69bc25ae15108a5',
+        'md5': 'f05d68a3769952c2d9257c473e14c15f',
+        'info_dict': {
+            'id': '55767ac51789ea11a94c0003ffd1fc09_2f007b0a37b94ec7a69bc25ae15108a5',
+            'ext': 'm4a',
+            'track': 'insame',
+            'like_count': int,
+            'duration': 84.03,
+            'title': 'insame',
+            'view_count': int,
+            'comment_count': int,
+            'uploader': 'Sorakime',
+            'uploader_id': 'sorakime',
+            'thumbnail': 'https://bandlabimages.azureedge.net/v1.0/users/572a351a-0f3a-4c6a-ac39-1a5defdeeb1c/',
+            'timestamp': 1691162128,
+            'upload_date': '20230804',
+            'media_type': 'track',
+        },
+    }, {
+        'url': 'https://www.bandlab.com/revision/014de0a4-7d82-ea11-a94c-0003ffd19c0f',
+        'only_matching': True,
+    }]
+    _WEBPAGE_TESTS = [{
+        'url': 'https://phantomluigi.github.io/',
+        'info_dict': {
+            'id': 'e14223c3-7871-ef11-bdfd-000d3a980db3',
+            'ext': 'm4a',
+            'view_count': int,
+            'upload_date': '20240913',
+            'uploader_id': 'phantommusicofficial',
+            'timestamp': 1726194897,
+            'uploader': 'Phantom',
+            'comment_count': int,
+            'genres': ['Progresive Rock'],
+            'description': 'md5:a38cd668f7a2843295ef284114f18429',
+            'duration': 225.23,
+            'like_count': int,
+            'title': 'Vermilion Pt. 2 (Cover)',
+            'track': 'Vermilion Pt. 2 (Cover)',
+            'thumbnail': 'https://bandlabimages.azureedge.net/v1.0/songs/62b10750-7aef-4f42-ad08-1af52f577e97/',
+            'media_type': 'revision',
+        },
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+
+        qs = parse_qs(url)
+        revision_id = traverse_obj(qs, (('revId', 'id'), 0, any))
+        if 'bandlab.com/revision/' in url:
```

Comment:
not sure if it would be worth it, but alternatively we could add something like a `url_type` capture group to the `_VALID_URL` patterns